### PR TITLE
docs: instructions for building and running using docker locally

### LINKED ISSUE #163: Building and Testing with Docker Locally
Bug: When building + running with Docker locally (i.e. per service) one might get this error: 

```
google.auth.exceptions.DefaultCredentialsError: Could not automatically determine credentials. Please set GOOGLE_APPLICATION_CREDENTIALS or explicitly create credentials and re-run the application. For more information, please see https://cloud.google.com/docs/authentication/getting-started
```
For me, using the link to set up authentication and setting the env variable `GOOGLE_APPLICATION_CREDENTIALS` didn't fix this issue.

Solution:
1. build and tag the image `docker build . --tag gcr.io/[PROJECT-ID]/[IMAGE]`
2. configure docker (.docker/config.json file) using `gcloud auth configure-docker`
3. when using `docker run`, pass in some extra flags:
```
PORT=8080 && docker run \
-p 9090:${PORT} \
-e PORT=${PORT} \
-e K_SERVICE=dev \
-e K_CONFIGURATION=dev \
-e K_REVISION=dev-00001 \
-e GOOGLE_APPLICATION_CREDENTIALS=/tmp/keys/FILE_NAME.json \
-v $GOOGLE_APPLICATION_CREDENTIALS:/tmp/keys/FILE_NAME.json:ro \
gcr.io/PROJECT_ID/IMAGE
```

The GCP [docs](https://cloud.google.com/run/docs/building/containers) helped.

This doesn't seem to be an issue when using GKE since it automatically sets service account, so it's not an issue when deploying using the Makefile/current docs. Could be helpful to add a section to CONTRIBUTING.md.

### PR DESCRIPTION
**Pull request by gzhao408**
Fixes #163.

Adds instructions to CONTRIBUTING.md for building and running individual services using Docker locally. Addresses a potential issue if the local environment isn't set up to authenticate for GCP.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -242,3 +242,50 @@ INGRESS_HOST="$(kubectl -n istio-system get service istio-ingressgateway -o json
 echo "$INGRESS_HOST"
 curl -v "http://$INGRESS_HOST"
 ```
+
+### (Optional) Building and Running Individual Services Locally
+
+Each service runs in its own docker container; you can view the container images pushed to your GCP project's [Container Registry](https://pantheon.corp.google.com/gcr/images/). Instead of using something like GCP Cloud Build, you can also build and run each container locally.
```
Comment (simonz130) [inline CONTRIBUTING.md:248]: please change pantheon internal links to console.cloud.google.com counterparts (available externally).
We should not reference internal links on github

Comment (simonz130) [inline CONTRIBUTING.md:248]: Please add a note on when would you use this option - i. e. quick repeatable debugging of a single microservice app.

Comment (gzhao408) [reply CONTRIBUTING.md]: Ah, my bad. Replaced.

Comment (gzhao408) [reply CONTRIBUTING.md]: Added "recommended" note at the top.

**Pull request merged**